src/plugins/dashboard/public/services/dashboard_content_management/lib/check_for_duplicate_dashboard_title.ts

```diff
@@ -44,21 +50,38 @@ export async function checkForDuplicateDashboardTitle(
     return true;
   }
 
+  const [baseDashboardName, duplicationId] = extractTitleAndCount(title);
+
   const { hits } = await contentManagement.client.search<
     DashboardCrudTypes['SearchIn'],
     DashboardCrudTypes['SearchOut']
   >({
     contentTypeId: DASHBOARD_CONTENT_ID,
     query: {
-      text: title ? `${title}*` : undefined,
-      limit: 10,
+      text: title ? `${baseDashboardName}*` : undefined,
+      limit: searchLimit,
+      cursor: String(duplicationId ? Math.ceil(duplicationId / searchLimit) : 1),
```

Comment:
Do we need a cursor still?

Reply:
Not necessarily. Setting the cursor is only useful if there are more than the 20 duplicate dashboards, but even without this, we'll still come up with a good enough suggestion.